bolt-modules/boltlib/lib/puppet/functions/run_plan.rb

```diff
@@ -169,6 +178,16 @@ def extract_parameter_types(type_expr)
     end
   end
 
+  # Determine if the parameter is Sensitive and should be wrapped.
+  def sensitive?(type_expr)
+    # Parameterized Sensitive type (e.g. Sensitive[String])
+    (defined?(type_expr.left_expr) && type_expr.left_expr.cased_value == 'Sensitive') ||
```

Comment:
This feels like the right level of complexity to handle for types. I think we should add a check where if all of these are false and `type_expr.to_s.include?('Sensitive')` is true we raise warning that we can't handle complex types that include sensitive (for example `Variant[Sensitive[String], Array]`).

Reply:
Playing around with this I realized that we probably _don't_ want to warn when using a complex type with `Sensitive`. 

Complex data types using `Sensitive` _are_ supported by `run_plan()`, just not by `bolt plan run`. Something like this works since the `outer` plan is able to wrap the parameter as `Sensitive` before passing it to the `inner` plan:

```
plan inner(
  Variant[Sensitive[String], Array[String]] $foo
) {
  return $foo.unwrap
}

plan outer(
  String $foo
) {
  $result = run_plan('inner', 'foo' => Sensitive($foo))
  return $result
}

$ bolt plan run outer foo=foo
```

I've updated the function to not automatically wrap a parameter if it's _already_ `Sensitive`, since I didn't take this into account originally. But I'm not really sure if there's a way for us to easily distinguish when a complex data type using `Sensitive` is and is not supported, or if it's desirable in general. Maybe it's more of a documentation issue?

Reply:
Playing around with this I realized that we probably _don't_ want to warn when using a complex type with `Sensitive`. 

Complex data types using `Sensitive` _are_ supported by `run_plan()`, just not by `bolt plan run`. Something like this works since the `outer` plan is able to wrap the parameter as `Sensitive` before passing it to the `inner` plan:

```
plan inner(
  Variant[Sensitive[String], Array[String]] $foo
) {
  return $foo.unwrap
}

plan outer(
  String $foo
) {
  $result = run_plan('inner', 'foo' => Sensitive($foo))
  return $result
}

$ bolt plan run outer foo=foo
```

I've updated the code to not automatically wrap a parameter that is _already_ a `Sensitive` type, since I didn't take that into account previously.

I'm not really sure if there's a way for us to easily distinguish when a complex data type using `Sensitive` is and is not supported, or if it's desirable in general. Maybe it's more of a documentation issue?